src/Oceananigans.jl

```diff
@@ -162,9 +165,65 @@ const defaults = Defaults()
     AbstractModel
 
 Abstract supertype for models.
+
+Models are required to have the following fields:
+  - `model.clock::Clock`
+  - `model.architecture`
+  - `model.timestepper` with `timestepper.G⁻` and `timestepper.Gⁿ`
 """
 abstract type AbstractModel{TS, A} end
 
+# Method interface for AbstractModel
+
+"""
+    $SIGNATURES
+
+Returns the current iteration of the `model.clock`.
+"""
+iteration(model::AbstractModel) = model.clock.iteration
+
+"""
+    $SIGNATURES
+
+TODO: what does this method do in addition to the constructor?
+"""
+initialize!(model::AbstractModel) = nothing
+
+"""
+    $SIGNATURES
+
+TODO
+"""
+total_velocities(model::AbstractModel) = nothing
```

Comment:
Not really well defined at the moment. It returns the sum of background + prognostic velocities for the non hydrostatic model and the prognostic velocities in the hydrostatic model

Reply:
Updated in [be79014](https://github.com/CliMA/Oceananigans.jl/pull/4798/commits/be790147d1058a632b8721df37172ae6ba676c8d)

Reply:
I agree, but I didn't want to rock the boat too much.